### ISSUE TITLE
zeromq:x64-windows Error: BUILD_FAILED build failure

### ISSUE THREAD

**Issue by catscribbles** [microsoft/vcpkg#18582]
[config-x64-windows-out.log.txt](https://github.com/microsoft/vcpkg/files/6692330/config-x64-windows-out.log.txt)
**Host Environment**
- OS: windows 10
- Compiler: Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30038.1 for x64

**To Reproduce** 
From a x64 native tools command prompt
vcpkg install zeromq:x64-windows

**Failure logs** 
C:\Users\xxxxxxx\source\repos\vcpkg-master>vcpkg install zeromq:x64-windows
Computing installation plan...
The following packages will be built and installed:
    zeromq[core]:x64-windows -> 4.3.4
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\xxxxxxx\AppData\Local\vcpkg\archives\e2\e2bce5c57c93c90a638d2d8bfe44d51d19960103.zip
Starting package 1/1: zeromq:x64-windows
Building package zeromq[core]:x64-windows...
-- Using C:/Users/xxxxxxx/source/repos/vcpkg-master/downloads/zeromq-libzmq-v4.3.4.tar.gz
-- Cleaning sources at C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/src/v4.3.4-ea6f146646.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/xxxxxxx/source/repos/vcpkg-master/downloads/zeromq-libzmq-v4.3.4.tar.gz
-- Using source at C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/src/v4.3.4-ea6f146646.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: ninja -v
    Working Directory: C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Users\xxxxxxx
[config-x64-windows-out.log.txt](https://github.com/microsoft/vcpkg/files/6692315/config-x64-windows-out.log.txt)
\source\repos\vcpkg-master\buildtrees\zeromq\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:323 (vcpkg_execute_required_process)
  ports/zeromq/portfile.cmake:25 (vcpkg_configure_cmake)
  scripts/ports.cmake:139 (include)

Error: Building package zeromq:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: zeromq:x64-windows
  Vcpkg version: 2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9

Additionally, attach any relevant sections from the log files above.
**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

Any help would be greatly appreciated

***
zeromq\config-x86-windows-out.log
[1/2] cmd /c "cd ../../x64-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/src/v4.3.4-ea6f146646.clean" "-DZMQ_BUILD_TESTS=OFF" "-DBUILD_STATIC=0" "-DBUILD_SHARED=1" "-DWITH_PERF_TOOL=OFF" "-DWITH_DOCS=OFF" "-DWITH_NSS=OFF" "-DWITH_LIBSODIUM_STATIC=0" "-DWITH_LIBSODIUM=OFF" "-DENABLE_DRAFTS=OFF" "-DDISABLE_WS=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/xxxxxxx/source/repos/vcpkg-master/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/xxxxxxx/source/repos/vcpkg-master/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Users/xxxxxxx/source/repos/vcpkg-master" "-D_VCPKG_INSTALLED_DIR=C:/Users/xxxxxxx/source/repos/vcpkg-master/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_PDB_OUTPUT_DIRECTORY=C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/x64-windows-dbg" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/Users/xxxxxxx/source/repos/vcpkg-master/packages/zeromq_x64-windows/debug""
FAILED: ../../x64-windows-dbg/CMakeCache.txt 
cmd /c "cd ../../x64-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/src/v4.3.4-ea6f146646.clean" "-DZMQ_BUILD_TESTS=OFF" "-DBUILD_STATIC=0" "-DBUILD_SHARED=1" "-DWITH_PERF_TOOL=OFF" "-DWITH_DOCS=OFF" "-DWITH_NSS=OFF" "-DWITH_LIBSODIUM_STATIC=0" "-DWITH_LIBSODIUM=OFF" "-DENABLE_DRAFTS=OFF" "-DDISABLE_WS=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/xxxxxxx/source/repos/vcpkg-master/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/xxxxxxx/source/repos/vcpkg-master/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Users/xxxxxxx/source/repos/vcpkg-master" "-D_VCPKG_INSTALLED_DIR=C:/Users/xxxxxxx/source/repos/vcpkg-master/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_PDB_OUTPUT_DIRECTORY=C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/x64-windows-dbg" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/Users/xxxxxxx/source/repos/vcpkg-master/packages/zeromq_x64-windows/debug""
-- The C compiler identification is MSVC 19.29.30038.1
-- The CXX compiler identification is MSVC 19.29.30038.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - works
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE  
-- Detected ZMQ Version - 4.3.4
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Not building draft classes and methods
-- Using builtin sha1
-- Using tweetnacl for CURVE security
-- Detected _WIN32_WINNT from CMAKE_SYSTEM_VERSION: 0x0A00
-- Using polling method in I/O threads: epoll
-- Including wepoll
-- Using polling method in zmq_poll(er)_* API: poll
-- Using 64 bytes alignment for lock-free data structures
-- Looking for include file windows.h
-- Looking for include file windows.h - not found
-- Looking for include files winsock2.h, afunix.h
-- Looking for include files winsock2.h, afunix.h - found
-- Looking for include file condition_variable
-- Looking for include file condition_variable - found
-- Using condition_variable_t implementation: stl11
-- Looking for if_nametoindex
-- Looking for if_nametoindex - not found
-- Looking for SO_PEERCRED
-- Looking for SO_PEERCRED - not found
-- Looking for LOCAL_PEERCRED
-- Looking for LOCAL_PEERCRED - not found
CMake Error at CMakeLists.txt:571 (message):
  Cannot link to ws2_32 or ws2

-- Configuring incomplete, errors occurred!
See also "C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/x64-windows-dbg/CMakeFiles/CMakeOutput.log".
See also "C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/x64-windows-dbg/CMakeFiles/CMakeError.log".
[2/2] cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/src/v4.3.4-ea6f146646.clean" "-DZMQ_BUILD_TESTS=OFF" "-DBUILD_STATIC=0" "-DBUILD_SHARED=1" "-DWITH_PERF_TOOL=OFF" "-DWITH_DOCS=OFF" "-DWITH_NSS=OFF" "-DWITH_LIBSODIUM_STATIC=0" "-DWITH_LIBSODIUM=OFF" "-DENABLE_DRAFTS=OFF" "-DDISABLE_WS=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/xxxxxxx/source/repos/vcpkg-master/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/xxxxxxx/source/repos/vcpkg-master/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Users/xxxxxxx/source/repos/vcpkg-master" "-D_VCPKG_INSTALLED_DIR=C:/Users/xxxxxxx/source/repos/vcpkg-master/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/Users/xxxxxxx/source/repos/vcpkg-master/packages/zeromq_x64-windows""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/src/v4.3.4-ea6f146646.clean" "-DZMQ_BUILD_TESTS=OFF" "-DBUILD_STATIC=0" "-DBUILD_SHARED=1" "-DWITH_PERF_TOOL=OFF" "-DWITH_DOCS=OFF" "-DWITH_NSS=OFF" "-DWITH_LIBSODIUM_STATIC=0" "-DWITH_LIBSODIUM=OFF" "-DENABLE_DRAFTS=OFF" "-DDISABLE_WS=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/xxxxxxx/source/repos/vcpkg-master/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/xxxxxxx/source/repos/vcpkg-master/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Users/xxxxxxx/source/repos/vcpkg-master" "-D_VCPKG_INSTALLED_DIR=C:/Users/xxxxxxx/source/repos/vcpkg-master/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/Users/xxxxxxx/source/repos/vcpkg-master/packages/zeromq_x64-windows""
-- The C compiler identification is MSVC 19.29.30038.1
-- The CXX compiler identification is MSVC 19.29.30038.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE  
-- Detected ZMQ Version - 4.3.4
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Not building draft classes and methods
-- Using builtin sha1
-- Using tweetnacl for CURVE security
-- Detected _WIN32_WINNT from CMAKE_SYSTEM_VERSION: 0x0A00
-- Using polling method in I/O threads: epoll
-- Including wepoll
-- Using polling method in zmq_poll(er)_* API: poll
-- Using 64 bytes alignment for lock-free data structures
-- Looking for include file windows.h
-- Looking for include file windows.h - found
-- Looking for include files winsock2.h, afunix.h
-- Looking for include files winsock2.h, afunix.h - not found
-- Looking for include file condition_variable
-- Looking for include file condition_variable - not found
-- Using condition_variable_t implementation: win32api
-- Looking for WSAStartup
-- Looking for WSAStartup - found
-- Looking for UuidCreateSequential
-- Looking for UuidCreateSequential - found
-- Looking for GetAdaptersAddresses
-- Looking for GetAdaptersAddresses - not found
-- Looking for if_nametoindex
-- Looking for if_nametoindex - not found
-- Looking for fopen in ws2
-- Looking for fopen in ws2 - not found
CMake Error at CMakeLists.txt:579 (message):
  Cannot link to iphlapi

-- Configuring incomplete, errors occurred!
See also "C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/x64-windows-rel/CMakeFiles/CMakeOutput.log".
See also "C:/Users/xxxxxxx/source/repos/vcpkg-master/buildtrees/zeromq/x64-windows-rel/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.

**Comment by catscribbles** [comment]
Hi JonLiu,
I've not registered on github before, are you able to give me an idea of what I need to do to try to get the zeromq library built?
Thanks
Richard

**Comment by JonLiu1993** [comment]
@catscribbles ,Thanks for posting this issue,I cannot reproduce your issue locally.
````
if(WIN32 AND NOT CYGWIN)
  if(NOT HAVE_WS2_32 AND NOT HAVE_WS2)
    message(FATAL_ERROR "Cannot link to ws2_32 or ws2")
  endif()
````
According to the error, you are not linked to WS2_32.Lib. You can find it in the Windows SDK. Have you installed visual studio on your computer?

**Comment by catscribbles** [comment]
Hi JonLiu,
I have visual studio installed and updated with the latest update. I've noticed a few times when building within visual studio that on occasion the linker cannot be found and other files in my project folder cannot be found, sometimes there is an 'open in another process message'. I wait a second and retry the build and it goes through okay. I tried this with the zeromq build and on the third attempt it worked and I am able to build programs using the library. I wonder if it's not a compiler issue but it's something to do with antivirus? Anyway thanks for replying.
